src/gui/touchscreengui.cpp

```diff
@@ -107,6 +107,9 @@ static EKEY_CODE id_to_keycode(touch_gui_button_id id)
 			break;
 	}
 	assert(!key.empty());
+	if (g_settings->get("keymap_" + key) == "")
+		return KEY_NONE;
```

Comment:
Apparently `KEY_NONE` is mapped to a key on some systems:

```cpp
        KEY_NONE             = 0xFF,  // usually no key mapping, but some laptops use it for fn key
```

https://github.com/minetest/irrlicht/blob/03dd8b0336d041726dbec5e47dc8bfcbcf858767/include/Keycodes.h#L180

It would be better to use `KEY_KEY_CODES_COUNT` to mean "no key".

Reply:
I would suggest option `KEY_NONE` since it's kinda more descriptive? What do you think?

Reply:
Well, using `KEY_NONE` to mean "no key" would introduce a bug: If someone were to assign `KEY_NONE` to one of the TouchScreenGUI buttons, the button would be hidden. So I'd say we can't use it.

Also, we're already using `KEY_KEY_CODES_COUNT` to mean "no key" in `src/client/keycode.cpp`:

https://github.com/minetest/minetest/blob/2c44620e5e344ca2b4acaaaa5408502332ca5aff/src/client/keycode.cpp#L314-L320